### PR TITLE
add two config files for two canary jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
/area jobs
/hold
gonna experiment something

### CODE REVIEW

File: jobs/kubernetes/sig-testing/canaries.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - --scenario=kubernetes-e2e
+      - --
+      - --check-leaked-resources
+      - --cluster=canary-e2e-prow
+      - --extract=ci/latest
+      - --gcp-zone=us-central1-f
+      - --ginkgo-parallel=25
+      - --provider=gce
+      - --test_args=--ginkgo.skip=\\[Slow\\]|\\[Serial\\]|\\[Disruptive\\]|\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8
+      - --timeout=65m
```
Comment (BenTheElder) [inline jobs/kubernetes/sig-testing/canaries.yaml]: nit: newlines missing at the end of these

**Comment by krzyzacy** [comment]
/assign @BenTheElder @fejta 
this should be orthogonal to the updateconfig plugin implementation, I'd start with a handcraft configmap and make sure the configmap actually will work first

File: jobs/kubernetes/sig-testing/canaries.yaml (hunk 1 above)
Comment (krzyzacy) [reply jobs/kubernetes/sig-testing/canaries.yaml]: done

File: config/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Kubernetes Prow Job Configs
```
Comment (BenTheElder) [inline config/README.md]: I think this dir should be kubernetes project configs for the tools in general (with subdirs), it looks like that's what is happening but not what this doc says.

Comment (krzyzacy) [reply config/README.md]: not tools in general right?

Comment (krzyzacy) [reply config/README.md]: ok so in the future we'll move things like gubernator, velodrome deployment files here?

Comment (BenTheElder) [reply config/README.md]: yeah non-portable configs ought to be siloed away from the portable/usable code

Comment (krzyzacy) [reply config/README.md]: ok edited this one, and added prowjobs/README.md

**Comment by BenTheElder** [comment]
SGTM 👍

**Comment by krzyzacy** [comment]
hummm @fejta why? istio config should not live under this repo?

was talking with @BenTheElder, everything under k8s/test-infra/config is config related to k8s test-infra, aka:
```
config/
    prowjobs/
        kubernetes/
            sig-foo/
                foo.yaml
        kubeflow/
            kubeflow-tests.yaml
    gubernator/
        guber-config.yaml
    prow/
        deployments.yaml
    boskos/
        more-deployments.yaml
```

**Comment by fejta** [comment]
```
config/
  kubeflow/
    testgrid.yaml
    prowjobs.yaml
  kubernetes/
    sig-foo/
      prowjobs.yaml
      testgrid.yaml
    sig-bar/
      prowjobs.yaml
      testgrid.yaml
  istio/
    testgrid.yaml
```

**Comment by BenTheElder** [comment]
@krzyzacy I mean the exact structure within doesn't matter too much, my thought was just to over time move non-portable config bits to some dir `/config/` and their tests etc so that `prow/`, `testgrid/` etc are generic.

putting the github.com/kubernetes/* bits in some subdir of that sounds fine.

**Comment by krzyzacy** [comment]
@fejta how'd you filter out all the yaml files contain prowjobs?

**Comment by fejta** [comment]
> how'd you filter out all the yaml files contain prowjobs?

This is why I do not like the solution of putting the complexity of finding and parsing config split into multiple files inside of the prow runtime libraries, as opposed to continuing to have prow expect a single --config-path and then generating this giant file using whatever complexity we need.

**Comment by krzyzacy** [comment]
@fejta unfortunately we have never taken advantage from that testgrid feature :-)

**Comment by fejta** [comment]
I started to make a PR that uses it but got over excited and tried to split apart things more than necessary. The problem with testgrid is that the jobs aren't owned by sigs, so sig-foo/testgrid.yaml doesn't make sense. I should have just started with a `istio/testgrid.yaml` and `kubeflow/testgrid.yaml`

**Comment by krzyzacy** [comment]
also I feel like the top-level subdir should be grouped by product/tools we are hosting

where will you put config/prow/deployment-foo.yaml?

**Comment by fejta** [comment]
Sure SGTM

File: config/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package(default_visibility = ["//visibility:public"])
+
+filegroup(
+    name = "prowjobs",
+    srcs = glob([
+        "jobs/**/*.yaml",
```
Comment (krzyzacy) [inline config/BUILD.bazel]: hummm I wonder later on if this can filter out testgrid.yaml

**Comment by krzyzacy** [comment]
wanna re-stamp and :shipit: ? I'll make follow up PRs.

**Comment by krzyzacy** [comment]
/hold cancel
🤞

**Pull request merged**